Sources/WebUI/EnhancedWKWebView.swift

```diff
@@ -64,15 +64,23 @@ class EnhancedWKWebView: WKWebView {
         }
 
         override func responds(to aSelector: Selector!) -> Bool {
-            super.responds(to: aSelector) || delegate?.responds(to: aSelector) == true
+            let _delegate = MainActor.assumeIsolated {
+                delegate
+            }
+
+            return super.responds(to: aSelector) || _delegate?.responds(to: aSelector) == true
         }
 
         override func forwardingTarget(for aSelector: Selector!) -> Any? {
-            guard let delegate, delegate.responds(to: aSelector) else {
+            let _delegate = MainActor.assumeIsolated {
+                delegate
+            }
+
+            guard let _delegate, _delegate.responds(to: aSelector) else {
                 return super.forwardingTarget(for: aSelector)
             }
 
-            return delegate
+            return _delegate
```

Comment:
Oh, I had not thought of that idea.
That implementation certainly looks good, so I will change the implementation.

Reply:
Thanks for the changes.
In my opinion, since `endRefreshing()` is called only from methods within the extension, it might be better to move it to the extension implementation.

Reply:
I changed it. Please check it out.